lib/swoosh/mailer.ex

```diff
@@ -98,26 +110,26 @@ defmodule Swoosh.Mailer do
   @doc """
   Parses the OTP configuration at compile time.
   """
-  def parse_config(mailer, opts) do
-    otp_app = Keyword.fetch!(opts, :otp_app)
+  def parse_static_config(mailer, module_config) do
+    otp_app = Keyword.fetch!(module_config, :otp_app)
     config = Application.get_env(otp_app, mailer, [])
-    adapter = opts[:adapter] || config[:adapter]
```

Comment:
removed and added a test to ensure raise on KeyError with `~r/:adapter not found/`
I was trying to catch this on compile time still, but given now users can pass adapter at runtime, I guess it's no longer necessary.